File: Métodos Numericos/Lalo/Gauss Seiden/GaussSeiden.py
def imprimirMatriz(a):
    print("\nMatriz ampliada:")
    for fila in a:
        for elem in fila:
            print(f"{elem:10.6f}", end=" ")
        print()

def gaussSeidel(a, x, n, eaMax):
    ea = [1.0 for _ in range(n)]

    
    for i in range(n):
        x[i] = 0.0

    while True:
        terminar = True
        for i in range(n):
            s = 0.0
            xv = x[i]
            for j in range(n):
                if i != j:
                    s += a[i][j] * x[j]
            x[i] = (a[i][n] - s) / a[i][i]

            if x[i] != 0.0:
                ea[i] = abs((xv - x[i]) / x[i])
            if ea[i] >= eaMax:
                terminar = False
        if terminar:
            break

def main():
    n = 3
    eaMax = 0.001

    a = [
        [-8, 1, -2, -20],
        [2, -6, -1, -38],
        [-3, -1, 7, -34]
    ]

    imprimirMatriz(a)

    x = [0.0 for _ in range(n)]
    gaussSeidel(a, x, n, eaMax)

    print("\nSolución aproximada con Gauss-Seidel:")
    for i in range(n):
        print(f"x{i+1} = {x[i]:.6f}")


main()

File: Métodos Numericos/Asignacion7.4_00000248336/src/main/java/Main/GaussSeiden.java
package Main;

import java.util.Scanner;

/**
 * @author Jesus Gammael Soto Escalante 00000248336
 */
public class GaussSeiden {

    // Método para leer la matriz ampliada
    public static double[][] leerEcuaciones(int n) {
        Scanner sc = new Scanner(System.in);
        double[][] a = new double[n][n+1];

        System.out.println("Ingresa los coeficientes y resultados del sistema:");
        for (int i = 0; i < n; i++) {
            System.out.println("Ecuación " + (i+1) + ":");
            for (int j = 0; j < n; j++) {
                System.out.print("Coeficiente x" + (j+1) + ": ");
                a[i][j] = sc.nextDouble();
            }
            System.out.print("Resultado: ");
            a[i][n] = sc.nextDouble();
        }
        return a;
    }

    // Método para imprimir la matriz ampliada
    public static void imprimirMatriz(double[][] a) {
        System.out.println("\nMatriz ampliada:");
        for (double[] fila : a) {
            for (double elem : fila) {
                System.out.printf("%10.6f ", elem);
            }
            System.out.println();
        }
    }

    // Método Gauss-Seidel
    public static void gaussSeidel(double[][] a, double[] x, int n, double eaMax) {
        double[] ea = new double[n];
        double s, xv;
        boolean terminar;

        // Inicialización
        for (int i = 0; i < n; i++) {
            x[i] = 0.0;
            ea[i] = 1.0;
        }

        while (true) {
            terminar = true;
            for (int i = 0; i < n; i++) {
                s = 0.0;
                xv = x[i];
                for (int j = 0; j < n; j++) {
                    if (i != j) {
                        s += a[i][j] * x[j];
                    }
                }
                x[i] = (a[i][n] - s) / a[i][i];

                if (x[i] != 0.0) {
                    ea[i] = Math.abs((xv - x[i]) / x[i]);
                }
                if (ea[i] >= eaMax) {
                    terminar = false;
                }
            }
            if (terminar) break;
        }
    }

    public static void main(String[] args) {
        int n = 3;
        double eaMax = 0.001;

        double[][] a = leerEcuaciones(n);
        imprimirMatriz(a);

        double[] x = new double[n];
        gaussSeidel(a, x, n, eaMax);

        System.out.println("\nSolución aproximada con Gauss-Seidel:");
        for (int i = 0; i < n; i++) {
            System.out.printf("x%d = %.6f\n", i+1, x[i]);
        }
    }
}
